### PR TITLE
1682 new design for the login and registration area

### LINKED ISSUE #1682: 🚀 [Refactor] - Implement a new design for the login and registration area.
## 🚀 Feature
Implement a new design for the login and registration area.

the graphic designs by @HardiHart are on app.zeplin.io. 

Put the designs into a template for Gradido, the basis is Bootstrap bootstrap-vue.

- welcome
- login
- register
- reset password
- activation

### PR DESCRIPTION
**Pull request by ogerly**
## 🍰 Pullrequest
New template design

### Issues
- fixes #1682 

### pre Merge
pleace merge bevor:
- #1900 

### Sequence of issues 
- #1882 
- #1883 
- #1908

### Sub Branch

Has no the sub branch #1858

### CODE REVIEW

File: frontend/src/layouts/AuthLayout_gdd.vue [hunk 1]
```diff
@@ -1,19 +1,181 @@
...
+    },
+  },
+  mounted() {
+    this.$nextTick(() => {
+      window.addEventListener('resize', this.onResize)
+    })
+  },
+  watch: {
+    windowWidth() {
+      window.addEventListener('resize', this.onResize)
```
Comment (Mogge) [inline frontend/src/layouts/AuthLayout_gdd.vue:125]: You really want to add the same event listener every time windowWidth changes? I am not sure about this.
I suggest to have `windowWitdh`as computed property. Then we must verify, that `windowWidth` is updated when `window.innerWidth` updates. If this is the case, then we do not need to watch on `windowWidth`

File: frontend/src/layouts/DashboardLayout_gdd.vue [hunk 2]
```diff
@@ -45,6 +45,7 @@ import { FadeTransition } from 'vue2-transitions'
 import CONFIG from '@/config'
 
 export default {
+  name: 'DashboardTemplate',
```
Comment (Mogge) [inline frontend/src/layouts/DashboardLayout_gdd.vue:48]:
```suggestion
  name: 'DashboardLayout',
```
Also consider to change the file name to `DashboardLayout.vue`

File: frontend/src/locales/de.json [hunk 3]
```diff
@@ -208,24 +223,29 @@
...
+    "forgotPassword": {
+      "heading": "Bitte gib deine E-Mail an mit der du bei Gradido angemeldet bist."
+    },
     "login": {
-      "community": "Tausend Dank, weil du bei uns bist!",
-      "heading": "Gradido",
-      "new_wallet": "Neues Konto erstellen",
-      "signin": "Anmelden"
+      "community": "Wähle deine Gemeinschaft",
+      "heading": "Melde dich mit deinen Zugangsdaten an. Bewahre sie stet's sicher auf!",
```
Comment (Mogge) [inline frontend/src/locales/de.json:234]:
```suggestion
      "heading": "Melde dich mit deinen Zugangsdaten an. Bewahre sie stets sicher auf!",
```

File: frontend/src/locales/de.json [hunk 4]
```diff
@@ -261,5 +281,7 @@
   "transaction-link": {
     "send_you": "sendet dir"
   },
-  "via_link": "über einen Link"
+  "via_link": "über einen Link",
+  "welcome": "Willkommen",
+  "|": " | "
```
Comment (Mogge) [inline frontend/src/locales/de.json:286]: math.pipe

File: frontend/src/locales/en.json [hunk 5]
```diff
@@ -1,6 +1,24 @@
...
+    "left": {
+      "gratitude": "Gratitude",
+      "hasAccount": "You already have an account?",
+      "hereLogin": "Log in here",
+      "learnMore": "Learn more...",
+      "newCurrency": "The new currency",
+      "oneAnotherNature": "For One Another, For All, For Nature"
+    },
+    "navbar": {
+      "overGradido": "Over Gradido"
```
Comment (Mogge) [inline frontend/src/locales/en.json:17]:
```suggestion
      "aboutGradido": "About Gradido"
```

File: frontend/src/pages/ForgotPassword.vue [hunk 6]
```diff
@@ -36,18 +24,10 @@
...
-        :subtitle="$t('site.thx.email')"
-        :buttonText="$t('login')"
-        linkTo="/login"
-      />
-      <message
-        v-else
-        :headline="$t('site.thx.errorTitle')"
-        :subtitle="$t('error.email-already-sent')"
+        :headline="success ? $t('site.thx.title') : $t('site.thx.errorTitle')"
+        :subtitle="success ? $t('site.thx.email') : $t('error.email-already-sent')"
```
Comment (Mogge) [inline frontend/src/pages/ForgotPassword.vue:30]: I like this!

**Review by Mogge — CHANGES_REQUESTED** [review]

**Comment by Tirokk** [comment]
I continue my review tomorrow morning …

File: frontend/src/assets/scss/custom/_variables.scss [hunk 7]
```diff
@@ -978,4 +978,4 @@ $btn-border-radius-sm: $input-border-radius !default;
 
 // Import Bootstrap variable defaults
 
-@import "~bootstrap/scss/variables";
+@import "../../../../node_modules/bootstrap/scss/variables";
```
[the remark this answers is not in the document]
Comment (ogerly) [reply frontend/src/assets/scss/custom/_variables.scss:981]: an alternative for **../../../** 
would be **~/**
but this does not work. 
i have no idea why
only **node_module/** does not work in any case

File: frontend/src/assets/scss/gradido.scss [hunk 8]
```diff
@@ -1,215 +1,54 @@
 // Bootstrap (4.5.3) functions
 
-@import "~bootstrap/scss/functions";
+@import "../../../node_modules/bootstrap/scss/functions";
```
[the remark this answers is not in the document]
Comment (ogerly) [reply frontend/src/assets/scss/gradido.scss:3]: an alternative for **../../../** 
would be **~/**
but this does not work. 
i have no idea why
only **node_module/** does not work in any case

File: frontend/src/components/Auth/Navbar.vue [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+        width="230"
+        alt="start background image"
+      ></b-img>
+      <b-img class="sheet-img position-absolute d-block d-lg-none zindex1000" :src="sheet"></b-img>
+      <b-navbar-toggle target="nav-collapse" class="zindex1000"></b-navbar-toggle>
+
+      <b-collapse id="nav-collapse" is-nav class="mt-5 mt-lg-0">
+        <b-navbar-nav class="ml-auto" right>
+          <b-nav-item href="https://gradido.net/de/" target="_blank">
+            {{ $t('auth.navbar.overGradido') }}
```
[the remark this answers is not in the document]
Comment (ogerly) [reply frontend/src/components/Auth/Navbar.vue:22]: changed

File: frontend/src/layouts/AuthLayout_gdd.spec.js [hunk 10]
```diff
@@ -1,5 +1,6 @@
-import { mount } from '@vue/test-utils'
+import { mount, RouterLinkStub } from '@vue/test-utils'
 import AuthLayoutGdd from './AuthLayout_gdd'
+import VueRouter from 'vue-router'
```
[the remark this answers is not in the document]
Comment (ogerly) [reply frontend/src/layouts/AuthLayout_gdd.spec.js:3]: deleted

File: frontend/src/layouts/AuthLayout_gdd.vue [hunk 11]
```diff
@@ -1,19 +1,181 @@
...
+    MobileStart,
+    Navbar,
+    NavbarSmall,
+    Carousel,
     LanguageSwitch,
+    AuthFooter,
+  },
+  data() {
+    return {
+      mobileStart: window.innerWidth < 1025,
```
[the remark this answers is not in the document]
Comment (ogerly) [reply frontend/src/layouts/AuthLayout_gdd.vue:103]: lg: 1025

File: frontend/src/layouts/AuthLayout_gdd.vue [hunk 12]
```diff
@@ -1,19 +1,181 @@
...
+      windowWidth: window.innerWidth,
+    }
+  },
+  methods: {
+    isMobileStart(boolean) {
```
[the remark this answers is not in the document]
Comment (ogerly) [reply frontend/src/layouts/AuthLayout_gdd.vue:108]: ok

File: frontend/src/layouts/AuthLayout_gdd.vue (hunk 1 above)
Comment (ogerly) [reply frontend/src/layouts/AuthLayout_gdd.vue:125]: this was also my idea at the beginning. but unfortunately i could not watch window.innerWidth

File: frontend/src/layouts/DashboardLayout_gdd.vue (hunk 2 above)
Comment (ogerly) [reply frontend/src/layouts/DashboardLayout_gdd.vue:48]: changed

File: frontend/src/components/Auth/Navbar.vue [hunk 13]
```diff
@@ -0,0 +1,69 @@
...
+          </b-nav-item>
+          <b-nav-item to="/register" class="authNavbar ml-lg-5">{{ $t('signup') }}</b-nav-item>
+          <span class="d-none d-lg-block mt-1">{{ $t('|') }}</span>
+          <b-nav-item to="/login" class="authNavbar">{{ $t('signin') }}</b-nav-item>
+        </b-navbar-nav>
+      </b-collapse>
+    </b-navbar>
+  </div>
+</template>
+<script>
```
Comment (Tirokk) [inline frontend/src/components/Auth/Navbar.vue:32]:
```suggestion
</template>

<script>
```

File: frontend/src/components/Auth/Navbar.vue [hunk 14]
```diff
@@ -0,0 +1,69 @@
+<template>
```
Comment (Tirokk) [inline frontend/src/components/Auth/Navbar.vue:1]: Navbar on mobil is not correct

File: frontend/src/components/Auth/NavbarSmall.vue [hunk 15]
```diff
@@ -0,0 +1,18 @@
...
+        <b-navbar-nav class="mx-auto mx-sm-auto mr-md-1">
+          <b-nav-item to="/register" class="authNavbar ml-lg-5">{{ $t('signup') }}</b-nav-item>
+          <span class="mt-1">{{ $t('|') }}</span>
+          <b-nav-item to="/login" class="authNavbar">{{ $t('signin') }}</b-nav-item>
+        </b-navbar-nav>
+      </b-collapse>
+    </b-navbar>
+  </div>
+</template>
+<script>
```
Comment (Tirokk) [inline frontend/src/components/Auth/NavbarSmall.vue:14]:
```suggestion
</template>

<script>
```

File: frontend/src/layouts/AuthLayout_gdd.vue [hunk 16]
```diff
@@ -1,19 +1,182 @@
...
+            </b-card>
+          </div>
+          <auth-footer v-if="!$route.meta.hideFooter" class="pr-5"></auth-footer>
+        </b-col>
+      </b-row>
+      <!-- <auth-layout-gdd />-->
     </div>
   </div>
 </template>
 <script>
```
Comment (Tirokk) [inline frontend/src/layouts/AuthLayout_gdd.vue:83]:
```suggestion
</template>

<script>
```

File: frontend/src/layouts/AuthLayout_gdd.vue [hunk 17]
```diff
@@ -1,19 +1,182 @@
...
+    })
+  },
+  watch: {
+    windowWidth() {
+      window.addEventListener('resize', this.onResize)
+    },
   },
 }
 </script>
+<style lang="scss">
```
Comment (Tirokk) [inline frontend/src/layouts/AuthLayout_gdd.vue:131]:
```suggestion
</script>

<style lang="scss">
```

File: frontend/src/locales/de.json (hunk 4 above)
Comment (Tirokk) [reply frontend/src/locales/de.json:286]: Just put the string `" | "` directly in the code.

If the linting fails just exclude it from linting!
I like it WYSIWYG!!!

File: frontend/src/locales/de.json [hunk 18]
```diff
@@ -1,6 +1,24 @@
...
+  "1000thanks": "1000 Dank, weil du bei uns bist!",
+  "125": "125%",
+  "85": "85%",
   "advanced-calculation": "Vorausberechnung",
+  "auth": {
+    "left": {
+      "gratitude": "Dankbarkeit",
+      "hasAccount": "Du hast schon einen Account?",
+      "hereLogin": "Hier Anmelden",
+      "learnMore": "Erfahre mehr...",
```
Comment (Tirokk) [inline frontend/src/locales/de.json:12]:
```suggestion
      "learnMore": "Erfahre mehr …",
```

File: frontend/src/locales/en.json [hunk 19]
```diff
@@ -1,6 +1,24 @@
...
+  "85": "85%",
   "advanced-calculation": "Advanced calculation",
+  "auth": {
+    "left": {
+      "gratitude": "Gratitude",
+      "hasAccount": "You already have an account?",
+      "hereLogin": "Log in here",
+      "learnMore": "Learn more...",
+      "newCurrency": "The new currency",
+      "oneAnotherNature": "For One Another, For All, For Nature"
```
Comment (Tirokk) [inline frontend/src/locales/en.json:14]:
```suggestion
      "oneAnotherNature": "FOR ONE ANOTHER, FOR ALL, FOR NATURE"
```

File: frontend/src/locales/en.json [hunk 20]
```diff
@@ -1,6 +1,24 @@
...
+  "1000thanks": "1000 thanks for being with us!",
+  "125": "125%",
+  "85": "85%",
   "advanced-calculation": "Advanced calculation",
+  "auth": {
+    "left": {
+      "gratitude": "Gratitude",
+      "hasAccount": "You already have an account?",
+      "hereLogin": "Log in here",
+      "learnMore": "Learn more...",
```
Comment (Tirokk) [inline frontend/src/locales/en.json:12]:
```suggestion
      "learnMore": "Learn more …",
```

**Review by Tirokk — DISMISSED** [review]
Wunderful work @ogerly 😍

File: frontend/src/components/Auth/Carousel.vue [hunk 21]
```diff
@@ -0,0 +1,28 @@
+<template>
+  <div>
+    <b-carousel :interval="30000">
+      <b-carousel-slide img-src="/img/template/Foto_01.jpg"></b-carousel-slide>
+      <b-carousel-slide img-src="https://picsum.photos/1324/2324/?image=50"></b-carousel-slide>
+      <b-carousel-slide img-src="https://picsum.photos/1024/2024/?image=58"></b-carousel-slide>
+      <b-carousel-slide img-src="https://picsum.photos/1024/2024/?image=83"></b-carousel-slide>
```
[the remark this answers is not in the document]
Comment (ogerly) [reply frontend/src/components/Auth/Carousel.vue:7]: is local im public

File: frontend/src/locales/de.json (hunk 4 above)
Comment (ogerly) [reply frontend/src/locales/de.json:286]: I understand ... we already have the pipe ... 
I overlooked that... 
I have replaced this

File: frontend/src/assets/scss/custom/_variables.scss (hunk 7 above)
[the remark this answers is not in the document]
Comment (ogerly) [reply frontend/src/assets/scss/custom/_variables.scss:981]: # ../../../../node_modules/
change to 

# **~**

File: frontend/src/assets/scss/gradido.scss (hunk 8 above)
[the remark this answers is not in the document]
Comment (ogerly) [reply frontend/src/assets/scss/gradido.scss:3]: # ../../../../node_modules/
change to 

# **~**

File: frontend/src/assets/scss/gradido-template.scss [hunk 22]
```diff
@@ -0,0 +1,243 @@
...
+  color: #b6d939;
+}
+
+.gradido-global-color-619d338 {
+  color: #8ebfb1;
+}
+
+.gradido-global-color-44819a9 {
+  color: #026873;
+}
```
[the remark this answers is not in the document]
Comment (ogerly) [reply frontend/src/assets/scss/gradido-template.scss:192]: deleted unused classes, 
refactor scss to bootstrap classes

File: frontend/src/layouts/AuthLayout_gdd.vue (hunk 1 above)
Comment (ogerly) [reply frontend/src/layouts/AuthLayout_gdd.vue]: is deleted

File: frontend/.eslintrc.js [hunk 23]
```diff
@@ -50,6 +50,7 @@ module.exports = {
           '/time./',
           '/decay.types./',
           'settings.password.resend_subtitle',
+          'settings.password.reset',
```
[the remark this answers is not in the document]
Comment (ogerly) [reply frontend/.eslintrc.js:53]: I am not quite sure now ... but can we delete the thx.vue ?

File: frontend/src/components/Auth/Footer.vue [hunk 24]
```diff
@@ -0,0 +1,52 @@
+<template>
```
[the remark this answers is not in the document]
Comment (ogerly) [reply frontend/src/components/Auth/Footer.vue:1]: ok

File: frontend/src/components/Auth/Navbar.vue (hunk 14 above)
Comment (ogerly) [reply frontend/src/components/Auth/Navbar.vue:1]: ok

File: frontend/src/pages/ForgotPassword.spec.js [hunk 25]
```diff
@@ -162,23 +152,9 @@ describe('ForgotPassword', () => {
...
-
-    describe('comingFrom login', () => {
-      beforeEach(() => {
-        wrapper = Wrapper(createMockObject('resetPassword'))
-      })
-
-      it('has another subtitle', () => {
-        expect(wrapper.find('p.text-lead').text()).toEqual('settings.password.resend_subtitle')
-      })
-    })
```
Comment (Tirokk) [inline frontend/src/pages/ForgotPassword.spec.js:182]: Why this is removed?

The code to extract the comming from is still in `ForgotPassword.vue`.

Comment (ogerly) [reply frontend/src/pages/ForgotPassword.spec.js:182]: '.text-lead' no longer exists in the file ForgotPassword.js

**Review by Hardy-Hart-am-Start — COMMENTED** [review]
Bitte die Korrekturliste Beachten, da bevor die Bilder und Texte für die Slide Seite nicht freigegeben sind, noch nicht veröffentlicht werden kann.
[Korrektur_21_05_2022.pdf](https://github.com/gradido/gradido/files/8747347/Korrektur_21_05_2022.pdf)

File: frontend/src/locales/de.json [hunk 26]
```diff
@@ -1,5 +1,22 @@
...
+  "85": "85%",
   "advanced-calculation": "Vorausberechnung",
+  "auth": {
+    "left": {
+      "gratitude": "Dankbarkeit",
+      "hasAccount": "Du hast schon einen Account?",
+      "hereLogin": "Hier Anmelden",
+      "learnMore": "Erfahre mehr …",
+      "newCurrency": "Die neue Währung",
+      "oneAnotherNature": "FÜR EINANDER, FÜR ALLE, FÜR DIE NATUR"
```
Comment (Mogge) [inline frontend/src/locales/de.json:14]: Ist Bernd damit einverstanden? Richtig wäre `Füreinander`

**Review by Mogge — APPROVED** [review]

File: frontend/src/locales/de.json (hunk 26 above)
Comment (Mogge) [reply frontend/src/locales/de.json:14]: :+1:

**Pull request merged**